src/Sarif.Driver/Sdk/MultithreadedAnalyzeCommandBase.cs

```diff
@@ -1075,6 +1075,21 @@ private static SupportedPlatform GetCurrentRunningOS()
                 ThrowExitApplicationException(ExitReason.NoRulesLoaded);
             }
 
+            ISet<string> disabledSkimmers = BuildDisabledSkimmersSet(context, skimmers);
+
+            if (disabledSkimmers.Count == skimmers.Count())
+            {
+                Errors.LogAllRulesExplicitlyDisabled(context);
+                ThrowExitApplicationException(ExitReason.NoRulesLoaded);
+            }
+
+            this.CheckIncompatibleRules(skimmers, context, disabledSkimmers);
+
+            MultithreadedAnalyzeTargets(context, skimmers, disabledSkimmers);
+        }
+
+        public static ISet<string> BuildDisabledSkimmersSet(TContext context, IEnumerable<Skimmer<TContext>> skimmers)
```

Comment:
@scottoneil-ms This is a user facing method now. There should be tests for it, and XML docs.